cl/phase1/network/services/BlsToExecutionChange.go

```diff
@@ -0,0 +1,110 @@
+package services
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/Giulio2002/bls"
+	libcommon "github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon/cl/beacon/beaconevents"
+	"github.com/ledgerwatch/erigon/cl/beacon/synced_data"
+	"github.com/ledgerwatch/erigon/cl/clparams"
+	"github.com/ledgerwatch/erigon/cl/cltypes"
+	"github.com/ledgerwatch/erigon/cl/fork"
+	"github.com/ledgerwatch/erigon/cl/pool"
+	"github.com/ledgerwatch/erigon/cl/utils"
+)
+
+type blsToExecutionChangeService struct {
+	operationsPool    pool.OperationsPool
+	emitters          *beaconevents.Emitters
+	syncedDataManager *synced_data.SyncedDataManager
+	beaconCfg         *clparams.BeaconChainConfig
+}
+
+func NewBLSToExecutionChangeService(
+	operationsPool pool.OperationsPool,
+	emitters *beaconevents.Emitters,
+	syncedDataManager *synced_data.SyncedDataManager,
+	beaconCfg *clparams.BeaconChainConfig,
+) BLSToExecutionChangeService {
+	return &blsToExecutionChangeService{
+		operationsPool:    operationsPool,
+		emitters:          emitters,
+		syncedDataManager: syncedDataManager,
+		beaconCfg:         beaconCfg,
+	}
+}
+
+func (s *blsToExecutionChangeService) ProcessMessage(ctx context.Context, subnet *uint64, msg *cltypes.SignedBLSToExecutionChange) error {
+	// https://github.com/ethereum/consensus-specs/blob/dev/specs/capella/p2p-interface.md#bls_to_execution_change
+	defer s.emitters.Publish("bls_to_execution_change", msg)
+	// [IGNORE] The signed_bls_to_execution_change is the first valid signed bls to execution change received
+	// for the validator with index signed_bls_to_execution_change.message.validator_index.
+	if s.operationsPool.BLSToExecutionChangesPool.Has(msg.Signature) {
```

Comment:
i copied and pasted the pool check, but feel like it doesn't meet the requirement description in spec. can you help to  check?